openpgp/keys_test.go

```diff
@@ -999,3 +1003,95 @@ func TestRevokeSubkeyWithConfig(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestEntityValidate(t *testing.T) {
+	// RSA
+	entity, err := NewEntity("Golang Gopher", "Test Key", "no-reply@golang.com", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = entity.Validate(time.Now())
+	if err != nil {
+		t.Fatal("Valid entity was marked as invalid: ", err)
+	}
+	// Corrupt public modulo n in primary key
+	primaryKey := entity.PrivateKey
+	n := primaryKey.PublicKey.PublicKey.(*rsa.PublicKey).N
+	primaryKey.PublicKey.PublicKey.(*rsa.PublicKey).N = new(big.Int).Add(n, big.NewInt(1))
```

Comment:
Very small detail, `n+1` is an even number, so it might get refused by something else (if there is ever a check in the future). Just a proposal; generate a set of malicious (valid) keys and replace parameters from there.